pkg/kubelet/dockershim/network/kubenet/kubenet_linux.go

```diff
@@ -448,13 +547,33 @@ func (plugin *kubenetNetworkPlugin) teardown(namespace string, name string, id k
 			errList = append(errList, err)
 		}
 	}
+
+	iplist, exists := plugin.podIPs[id]
```

Comment:
This is going to need locking to protect against concurrent access with other containers.